sdk/schemaregistry/azure-schemaregistry-avroserializer/README.md

```diff
@@ -0,0 +1,122 @@
+
+```Bash
+pip install azure-schemaregistry-avroserializer
+```
+
+### Prerequisites: 
+To use this package, you must have:
+* Azure subscription - [Create a free account][azure_sub]
+* Azure Schema Registry
+* Python 2.7, 3.5 or later - [Install Python][python]
+
+### Authenticate the client
+Interaction with Schema Registry Avro Serializer starts with an instance of SchemaRegistryAvroSerializer class. You need the endpoint, AAD credential and schema group name to instantiate the client object. 
+
+**Create client using the azure-identity library:**
+
+```python
+from azure.schemaregistry.serializer.avro_serializer import SchemaRegistryAvroSerializer
+from azure.identity import DefaultAzureCredential
+
+credential = DefaultAzureCredential()
+endpoint = '<< ENDPOINT OF THE SCHEMA REGISTRY >>'
+schema_group = '<< GROUP NAME OF THE SCHEMA >>'
+schema_registry_client = SchemaRegistryAvroSerializer(endpoint, credential, schema_group)
+```
+
+## Key concepts
+
+- Avro: Apache Avro™ is a data serialization system.
+
+## Examples
+
+The following sections provide several code snippets covering some of the most common Schema Registry tasks, including:
+
+- [Serialization](serialization)
+- [Deserialization](deserialization)
+
+### Serialization
+
+```python
+import os
+from azure.schemaregistry.serializer.avro_serializer import SchemaRegistryAvroSerializer
+from azure.identity import DefaultAzureCredential
+
+token_credential = DefaultAzureCredential()
+endpoint = os.environ['SCHEMA_REGISTRY_ENDPOINT']
+schema_group = "<your-group-name>"
+
+serializer = SchemaRegistryAvroSerializer(endpoint, token_credential, schema_group)
+
+schema_string = """
+{"namespace": "example.avro",
+ "type": "record",
+ "name": "User",
+ "fields": [
+     {"name": "name", "type": "string"},
+     {"name": "favorite_number",  "type": ["int", "null"]},
+     {"name": "favorite_color", "type": ["string", "null"]}
+ ]
+}"""
+
+with serializer:
+    dict_data = {"name": "Ben", "favorite_number": 7, "favorite_color": "red"}
+    encoded_bytes = serializer.serialize(dict_data, schema_string)
+```
+
+### Deserialization
+
+```python
+import os
+from azure.schemaregistry.serializer.avro_serializer import SchemaRegistryAvroSerializer
+from azure.identity import DefaultAzureCredential
+
+token_credential = DefaultAzureCredential()
+endpoint = os.environ['SCHEMA_REGISTRY_ENDPOINT']
+schema_group = "<your-group-name>"
+
+serializer = SchemaRegistryAvroSerializer(endpoint, token_credential, schema_group)
+
+with serializer:
+    encoded_bytes = b'<data_encoded_by_azure_schema_registry_avro_serializer>'
+    decoded_data = serializer.deserialize(encoded_bytes)
+```
+
+## Troubleshooting
+
+Azure Schema Registry Avro Serializer raise exceptions defined in [Azure Core][azure_core].
+
+## Next steps
+
+### More sample code
+
+Please find further examples in the [samples](./samples) directory demonstrating common Azure Schema Registry Avro Serializer scenarios.
```

Comment:
relative links should be absolute per Cala's recommendations (otherwise docs break when we render them out to various sources)